Add TLS/SSL usage description (#571)
http-party/http-server: README.md

Note the `?` at the end of the proxy URL. Thanks to [@houston3](https://github.com/houston3) for this clever hack!

> ## Using SSL

Reviewer: To be pedantic, I think we should say "TLS/SSL" here, since the user will technically be enabling TLS, even though people sometimes call it SSL.

Author: OK

First, make sure that you have `key.pem` and `cert.pem` files. You can generate them using this command:

``` sh

> openssl req -newkey rsa:2048 -new -nodes -x509 -days 3650 -keyout key.pem -out cert.pem

Reviewer: Is `openssl` built in to all operating systems? If not, we should mention how to install it, or provide a link, or at the very least mention that it is a requirement.

Author: Finished

```

You will be prompted with a few questions after entering the command. Use `127.0.0.1` as value for `Common name` if you want to be able to install the certificate in your OS's root certificate store or browser so that it is trusted.

> This generates a cert-key pair and it will be valid for roughly 10 years (3650 days to be exact).

Reviewer: What do you think of reversing the phrasing here to say

```suggestion
 This generates a cert-key pair and it will be valid for 3650 days (about 10 years).
```

Author: Finished

Then you need to run the server with `-S` for enabling SSL and `-C` for your certificate file:

``` sh

> http-server -S -C cert.pem -o

Reviewer: `-o` isn't relevant to TLS/SSL, so we shouldn't include it here.

Author: Finished

Starting up http-server, serving ./ through https
Available on:
  https:127.0.0.1:8080
  https:192.168.1.101:8080
  https:192.168.1.104:8080

> Hit CTRL-C to stop the server

Reviewer: Including this with the command may imply that these are also commands, when they are actually output. Maybe we should say "This is what should be output if successful"

Author: Finished

Reviewer: Looking back at my comment, I wasn't very clear about what I meant. I think it would be more clear to say something like:

    Then you need to run the server with `-S` for enabling SSL and `-C` for your certificate file.

    ```sh
    http-server -S -C cert.pem
    ```

    This is what should be output if successful:

    ```
    Starting up http-server, serving ./ through https
    Available on:
      https:127.0.0.1:8080
      https:192.168.1.101:8080
      https:192.168.1.104:8080
    Hit CTRL-C to stop the server
    ```

Reviewer: That way, users won't need to stop to think whether they need to copy-paste the "Starting up" output

Author: OK. It's better to understand.
